### PR TITLE
checkActivities helper function

### PR DESCRIPTION
**Pull request by tschuy**

### CODE REVIEW

**Comment by tschuy** [comment]
the tl;dr is that it will throw with a list of all of the failing slugs, or continue cleanly with a list of activity ids.

**Comment by ElijahCaine** [comment]
@tschuy +1 on the tests. We had a small meatspace conversation about them, I think they test enough use cases (and failure cases) to move forward.

File: tests/helpers.js [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+var helpers = require('../src/helpers');
 
 module.exports = function(expect) {
-    // This line exists to please the linters and tests
-    // Remove it when you write real tests.
-    expect(helper);
+    describe('checkActivities', function() {
+        it('returns a list of activities IDs for proper slugs', function(done) {
+            helpers.checkActivities(['docs', 'dev']).then(function(activities) {
+                expect(activities).to.deep.equal([1, 2]);
```
Comment (iankronquist) [inline tests/helpers.js]: Silly question, if the test Promise is rejected and the rejection is not caught, does the test fail?

Comment (tschuy) [inline tests/helpers.js]: Yes, since literally what Mocha does is catch errors.

File: tests/helpers.js [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+        it('throws when passed a list containing bad slugs', function(done) {
+            helpers.checkActivities(['docs', 'dev', 'cats', 'dogs']).then()
+            .catch(function(err) {
+                err.sort();
+                expect(err).to.deep.have.same.members(['cats', 'dogs'].sort());
+                done();
+            });
+        });
+
+        it('throws when passed a list containing a null slug', function(done) {
```
Comment (tschuy) [inline tests/helpers.js]: change this to not mention list and rebase

File: src/helpers.js [hunk 3]
```diff
@@ -86,3 +86,40 @@ module.exports = function(app) {
 
     return helpers;
 };
+// src/helpers.js
+
+var app = require('./app');
+var knex = app.get('knex');
+//var errors = require('./errors');
+
+module.exports = {
```
Comment (LyonesGamer) [inline src/helpers.js]: I think we might have had some sort of merge error here or something. Why isn't checkActivities() a part of helpers like the other functions in the file?

File: tests/helpers.js [hunk 4]
```diff
@@ -135,3 +135,40 @@ module.exports = function(expect, app) {
...
+        it('returns a list of activities IDs for proper slugs', function(done) {
+            helpers.checkActivities(['docs', 'dev']).then(function(activities) {
+                expect(activities).to.deep.have.same.members([1, 2]);
+                done();
+            });
+        });
+
+        it('throws when passed undefined', function(done) {
+            helpers.checkActivities(undefined).then().catch(function(err) {
+                expect(err).to.be.an('undefined');
```
Comment (tschuy) [inline tests/helpers.js]: We should change the return value to be explicit, like we did with checkProject: https://github.com/osuosl/timesync-node/blob/develop/tests/helpers.js#L117

**Comment by LyonesGamer** [comment]
@tschuy I made that fix and also merged the objects since checkActivities was kind of hanging out on its own. Does it look good?

**Comment by Kennric** [comment]
Looks good, nice and simple. +1

**Pull request merged**